addons/l10n_se_account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py

```diff
@@ -0,0 +1,99 @@
+from stdnum import luhn
+from odoo import models, _
+
+
+class AccountEdiXmlUBLBIS3(models.AbstractModel):
+    _inherit = 'account.edi.xml.ubl_bis3'
+
+    def _get_partner_party_tax_scheme_vals_list(self, partner, role):
+        # EXTENDS account.edi.xml.ubl_21
+        vals_list = super()._get_partner_party_tax_scheme_vals_list(partner, role)
+
+        # sources:
+        #  https://sfti.se/download/18.427140af179361c4e462cc34/1620641679827/Beskrivning%20av%20svenska%20valideringsregler%202018-11-12.pdf
+        #  https://docs.peppol.eu/poacc/billing/3.0/bis/#national_rules (SE-R-005 (fatal))
+        if partner.country_id.code == "SE" and role == 'supplier':
+            vals_list.append({
+                'company_id': "GODKÄND FÖR F-SKATT",
+                'tax_scheme_vals': {'id': 'TAX'},
+            })
+        return vals_list
+
+    def _export_invoice_vals(self, invoice):
+        # EXTENDS account.edi.xml.ubl_21
+        vals = super()._export_invoice_vals(invoice)
+
+        # Add RequisitionDocumentReference from invoice.ref if available
+        if invoice.ref:
+            vals['vals']['requisition_document_reference'] = invoice.ref
+        # Add order reference if invoice_origin is set
+        if invoice.invoice_origin:
+            vals['vals']['order_reference_vals'] = {'id': invoice.invoice_origin}
+
+        return vals
+
+    def _invoice_constraints_peppol_en16931_ubl(self, invoice, vals):
+        constraints = super()._invoice_constraints_peppol_en16931_ubl(invoice, vals)
+
+        if vals['supplier'].country_id.code == 'SE':
+            vat = vals['supplier'].vat.strip() if vals['supplier'].vat else ""
+            company_registration = vals['supplier'].company_registry.replace('-', '').strip() if vals['supplier'].company_registry else ""
+            tax_scheme = vals['supplier'].tax_scheme.strip().upper() if vals['supplier'].tax_scheme else ""
+            payment_means_code = vals['supplier'].payment_means_code.strip() if vals['supplier'].payment_means_code else ""
+            financial_branch_id = vals['supplier'].financial_branch_id.strip() if vals['supplier'].financial_branch_id else ""
+            account_id = vals['supplier'].account_id.strip() if vals['supplier'].account_id else ""
```

Comment:
Great initiative adding those constraints! But I don't think this code works as none of `tax_scheme`, `payment_means_code`, `financial_branch_id` and `account_id` are fields on `res.partner`.